### ISSUE TITLE
Wiring and 

### ISSUE THREAD

**Issue by soloam** [RoganDawes/ESPHome_Wintex#2]
Hello, can you provide a wiring diagram for the Texcom Premier 832? I can't seem to find an image showing the connection of the esp32 to the panel. 

Also are you able to capture the sensors inputs (like motion and flood) ? And are you able to use the esp to trigger an alarm?

Thank you

**Comment by RoganDawes** [comment]
The main thing is that you need to interface at 5V ttl levels with the panel, while the ESP32 is a 3v3 device.
I used a 4-channel level converter module (like an Adafruit kit), but there are definitely other ways to do it, eg resistor divider networks, etc.
You can also pretty much choose any pins for the two serial ports, because of the ESP32 multiplexer.
So a lot of this comes down to personal preference, and availability of hardware.

**Comment by soloam** [comment]
But where do you connect it in the panel? And what are you able to retrieve and to send!

**Comment by RoganDawes** [comment]
The panel has two serial ports in the top right hand corner that you can connect to.

this post has some more details on the actual hardware: https://universaldiscoverymethodology.com/2015/04/16/texecom-com-ip-controller-diy-and-cheap-too

And in theory, I am able to retrieve anything that the Wintex software can retrieve, and send anything that Wintex can send. Currently, my ESPHome implementation is merely pulling the zone status, so that I can see whether a zone is active or not - essentially presence detection. But I do have plans to pull more information from the panel, such as partition status, etc, as well as allowing remote bypass of zones, and remote activation/deactivation of the partitions.

**Comment by leocrawford** [comment]
With the level shifter, IIUC a level shifter needs a HV of 5v high applied to the top of the level shifter, but the alarm panel doesn't seem to have a 5V pin anywhere.  

My best guess is that is only required for the 3.3v->5v level shift (ESP is doing TX and alarm doing RX) so that could be wired directly, and simply not apply the HV to the level shifter. The LV can easily be taken off the ESP.  But I could easily be wrong, so any guidance required.

Out of interest, do you power the ESP off the alarm. If so is that a simple 12v -> 3/5v converter?

**Comment by RoganDawes** [comment]
Yes, I power my ESP32 from the alarm panel itself, using a switching DC-DC converter to get 5V, and then the ESP32 dev board has a LDO regulator to step the 5V down to 3V3.

**Comment by leocrawford** [comment]
Thanks, and you use the 5v you get as HV input to level shifter?

**Comment by RoganDawes** [comment]
Exactly!
